File: Swift/implement-strstr.swift
class Solution {
    func strStr(_ haystack: String, _ needle: String) -> Int {
        let hCount = haystack.count
        let nCount = needle.count
        
        if nCount == 0 {
            return 0
        }
        if hCount < nCount {
            return -1
        }
        
        for i in 0...hCount-nCount {
            let start = haystack.index(haystack.startIndex, offsetBy: i)
            let end = haystack.index(haystack.startIndex, offsetBy: i+nCount)
            if String(haystack[start..<end]) == needle {
                return i
            }
        }
        
        return -1
    }
}

File: Python/implement-strstr.py
class Solution:
    def strStr1(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        hLen = len(haystack)
        nLen = len(needle)
        
        if nLen == 0:
            return 0
        if hLen < nLen:
            return -1

        for i in range(hLen - nLen + 1):
            if haystack[i:i+nLen] == needle:
                return i
        return -1

    def strStr2(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        return haystack.find(needle)
